engine/src/main/java/com/datatorrent/stram/stream/OiOStream.java

```diff
@@ -29,11 +29,12 @@
  *
  * @since 0.3.5
  */
-public class OiOStream implements Stream, SweepableReservoir
```

Comment:
OiOStream has only one thread operating on it, why is there a need for separate counts or issue with reset?